### PR TITLE
Fixing the missing columns needed by BitmapIndexScan (DynamicBitmapScan)

### PR DESCRIPTION
**Pull request by vraghavan78**
The columns required in the bitmap recheck and filter condition where missing in the project list of the BitmapIndexScan (or DynamicBitmapScan). This can cause a crash or wrong results during query execution. This PR fixes this issue.

@xinzweb @hsyuan please take a look.

[#110297108]

### CODE REVIEW

File: libgpopt/include/gpopt/translate/CTranslatorExprToDXL.h [hunk 1]
```diff
@@ -783,6 +783,17 @@ namespace gpopt
 			static
 			EdxlSubPlanType Edxlsubplantype(CExpression *pexprCorrelatedNLJoin);
 
+			// add used columns in the bitmap re-check and remainder filter condition to the
```
Comment (xinzweb) [inline libgpopt/include/gpopt/translate/CTranslatorExprToDXL.h]: What's `remainder` filter condition? Is there an official term for it? Any example? Thanks.

File: libgpopt/include/gpopt/translate/CTranslatorExprToDXL.h [hunk 2]
```diff
@@ -783,6 +783,17 @@ namespace gpopt
...
+			// required output column
+			static
+			void AddBitmapFilterColumns
+				(
+				IMemoryPool *pmp,
+				CPhysicalScan *pop,
+				CExpression *pexprRecheckCond,
+				CExpression *pexprScalar,
```
Comment (xinzweb) [inline libgpopt/include/gpopt/translate/CTranslatorExprToDXL.h]: Are you referring to the `pexprScalar` as the `remainder` filter condition?

File: data/dxl/minidump/sql/DynamicBitmapIndexScan.sql [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+SELECT fid FROM
+(SELECT b1.flex_value_set_id, b1.flex_value_id
+FROM
+inner_table b1,
+outer_table c1
+WHERE
+b1.flex_value_id = c1.flex_value_id 
+and c1.language = 'ZHS') bar,
+foo
+WHERE foo.fid = bar.flex_value_set_id;
```
Comment (xinzweb) [inline data/dxl/minidump/sql/DynamicBitmapIndexScan.sql]: There is a strange tailing `0` at the end.

File: data/dxl/minidump/sql/BitmapIndexScan.sql [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+    summary_flag character varying(1) ,
+    start_date_active date ,
+    end_date_active date ,
+    parent_flex_value_low character varying(60) ,
+    parent_flex_value_high character varying(60) ,
+    structured_hierarchy_level numeric(15,0) ,
+    hierarchy_level character varying(30) ,
+    compiled_value_attributes character varying(2000) ,
+    value_category character varying(30) ,
+    attribute1 character varying(240) ,
```
Comment (xinzweb) [inline data/dxl/minidump/sql/BitmapIndexScan.sql]: Do we really need these attribute1..50? The test `DynamicBitmapIndex` didn't have these attributes.

File: libgpopt/src/translate/CTranslatorExprToDXL.cpp [hunk 5]
```diff
@@ -742,6 +742,62 @@ CTranslatorExprToDXL::PdxlnBitmapTableScan
 			);
 }
 
+
+//---------------------------------------------------------------------------
+//	@function:
+//		CTranslatorExprToDXL::AddBitmapFilterColumns
+//
+//	@doc:
+//		Add used columns in the bitmap recheck and remainder filter condition
```
Comment (xinzweb) [inline libgpopt/src/translate/CTranslatorExprToDXL.cpp]: Again, the "remainder" comment.

File: data/dxl/minidump/sql/DynamicBitmapIndexScan.sql (hunk 3 above)
Comment (hsyuan) [inline data/dxl/minidump/sql/DynamicBitmapIndexScan.sql]: means EOF.

File: libgpopt/include/gpopt/translate/CTranslatorExprToDXL.h [hunk 6]
```diff
@@ -783,6 +783,17 @@ namespace gpopt
...
+				CColRefSet *pcrsReqdOutput
```
Comment (xinzweb) [inline libgpopt/include/gpopt/translate/CTranslatorExprToDXL.h]: Is there anyway to indicate we have the parameter `in` or `out` or `inout`? In this case, only the last parameter is `inout`. Thanks.

File: data/dxl/minidump/BitmapIndexScan.mdp [hunk 7]
```diff
@@ -0,0 +1,1252 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+see sql/BitmapIndexScan.sql
```
Comment (xinzweb) [inline data/dxl/minidump/BitmapIndexScan.mdp]: CDATA to have character data without escaping it. `<![CDATA[` and `]]>`.

File: data/dxl/minidump/sql/BitmapIndexScan.sql (hunk 4 above)
Comment (vraghavan78) [inline data/dxl/minidump/sql/BitmapIndexScan.sql]: no, I was lazy :( will fix it.

File: libgpopt/include/gpopt/translate/CTranslatorExprToDXL.h (hunk 6 above)
Comment (vraghavan78) [inline libgpopt/include/gpopt/translate/CTranslatorExprToDXL.h]: will add a comment.

**Comment by hsyuan** [comment]
LGTM. Don't forget to bump the version.

File: data/dxl/minidump/BitmapIndexScan.mdp (hunk 7 above)
Comment (vraghavan78) [inline data/dxl/minidump/BitmapIndexScan.mdp]: It does not work. I am going to leave the sql file.

File: libgpopt/include/gpopt/translate/CTranslatorExprToDXL.h (hunk 1 above)
Comment (vraghavan78) [inline libgpopt/include/gpopt/translate/CTranslatorExprToDXL.h]: remainder is the filter condition that is beyond the index condition

**Comment by vraghavan78** [comment]
Thanks!

**Pull request closed without merging**